src/main/java/org/broadinstitute/hellbender/utils/pileup/PileupElement.java

```diff
@@ -387,7 +411,7 @@ public CigarElement getNextOnGenomeCigarElement() {
      * @return a CigarElement, or null if no such element exists
      */
     private CigarElement getNeighboringOnGenomeCigarElement(final Direction direction) {
```

Comment:
Wait a sec...this method `getNeighboringOnGenomeCigarElement()` behaves very differently from `getNeighborNotOnGenomeOperator()`, since it iterates until it finds an "on-genome" operator, whereas your new method does not iterate. This is a somewhat confusing state of affairs -- can you clean this up to increase clarity?